internal/controller/agentgateway_controller.go

```
@@ -0,0 +1,732 @@
+			return ctrl.Result{}, nil
+		}
+		log.Error(err, "Failed to get AgentGateway")
+		return ctrl.Result{}, err
+	}
+
+	log.Info("Reconciling AgentGateway",
+		"name", agentGateway.Name,
+		"namespace", agentGateway.Namespace,
+		"agentGatewayClass", agentGateway.Spec.AgentGatewayClassName)
+
+	// Check if this controller is process for this AgentGateway
+	process := r.shouldProcessAgentGateway(ctx, &agentGateway)
+
+	if !process {
+		log.Info("Controller is not responsible for this AgentGateway, skipping reconciliation")
+		return ctrl.Result{}, nil
+	}
+
+	// Get all exposed agents
+	exposedAgents, err := r.getExposedAgents(ctx)
+	if err != nil {
+		log.Error(err, "Failed to get exposed agents")
+		return ctrl.Result{}, err
+	}
+
+	// Create ConfigMap
+	configMapName := agentGateway.Name + "-krakend-config"
+	if err := r.ensureConfigMap(ctx, &agentGateway, configMapName, exposedAgents); err != nil {
+		log.Error(err, "Failed to ensure ConfigMap for KrakenD")
+		return ctrl.Result{}, err
+	}
+
+	// Create Deployment
+	deploymentName := agentGateway.Name
+	if err := r.ensureDeployment(ctx, &agentGateway, deploymentName, configMapName); err != nil {
+		log.Error(err, "Failed to ensure Deployment for KrakenD")
+		return ctrl.Result{}, err
+	}
+
+	// Create Service
+	if err := r.ensureService(ctx, &agentGateway); err != nil {
+		log.Error(err, "Failed to ensure Service for AgentGateway")
+		return ctrl.Result{}, err
+	}
+
+	return ctrl.Result{}, nil
+}
+
+// shouldProcessAgentGateway determines if this controller is responsible for the given AgentGateway
+func (r *AgentGatewayReconciler) shouldProcessAgentGateway(ctx context.Context, agentGateway *agentruntimev1alpha1.AgentGateway) bool {
+
+	// If no className specified, check for default AgentGatewayClass
+	var agentGatewayClassList agentruntimev1alpha1.AgentGatewayClassList
+	if err := r.List(ctx, &agentGatewayClassList); err != nil {
+		// If we can't list classes, don't process to avoid errors
```

err is dropped here. Better log it as a warning